macros/src/codegen.rs

```diff
@@ -717,7 +715,7 @@ fn generate_action(
         };
         quote! {
             // ACTION
-            let _data = self.context.#action_ident(#temporary_context_call #g_a_param) #action_await;
+            let _data = self.context.#action_ident(#temporary_context_call #g_a_param) #action_await .map_err(#error_type_name::ActionFailed)?;
```

Comment:
We don't want to use the `?` operator because we need to restore the internal state of the state machine if the action fails.

Reply:
Not sure I understand what is proposed here. 
How can we restore the internal state in a generic way? 
If we remove simply `?` here,  the state machine will end up transitioning to the out state. 
By keeping the `?` we stay do not transition (stay in the original state), and propagate the error to the caller of process_event().
Could you clarify please .

Reply:
Yes, that's correct